.github/workflows/microshed-ci.yml

```
@@ -17,14 +18,22 @@ env:
   LANG: en_US.UTF-8
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+        java: [ 11 ]
+      fail-fast: false
+      max-parallel: 4
+    name: Build ${{ matrix.java }}, ${{ matrix.os }}
+
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
```

We should replace this with a concrete version number for security
```suggestion
    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

---

Dependabot will take care of this for us in the future :)